src/app/courses/add-courses/courses-step.component.ts

```diff
@@ -88,12 +88,21 @@ export class CoursesStepComponent implements OnDestroy {
     this.steps[this.activeStepIndex].resources.splice(position, 1);
   }
 
-  addExam() {
-    this.coursesService.stepIndex = this.activeStepIndex;
-    if (this.activeStep.exam) {
-      this.router.navigate([ '/courses/update/exam/', this.activeStep.exam._id ]);
-    } else {
-      this.router.navigate([ '/courses/exam/' ]);
+  addExam(type) {
+    if (type === 'exam') {
```

Comment:
This should be refactored to reduce complexity.  I think you don't need to add the extra route, you can just pass the `type` function parameter as a route parameter.

Reply:
much cleaner